### PR TITLE
Working with elements directly without ref

### PR DESCRIPTION
**Pull request by mohamedGamalAbuGalala**

### CODE REVIEW

File: src/createShadowDOM.ts [hunk 1]
```diff
@@ -15,9 +15,10 @@ const createShadowDOM = (currentChild, pushShadowItem) => {
     setNewChild();
   }
 
+  // @Discuss: Why this is not like this? ==> const setNewChild = (pushShadowItem, validChildren) =>{} and make it in module level.
```
Comment (mohamedGamalAbuGalala) [inline src/createShadowDOM.ts:18]: @medhatdawoud 
What do you think? :thinking:

File: src/setHighlight.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 const setHighlight = (ref, isInitial, hideAfter) => {
+  //TODO: Should remove && ref.current after replacing children with childrenWithRef.
   if (!isInitial && ref.current) {
     const element = ref.current.style;
     element.background = "#f8ffb4";
+    // @Discuss: Why Not clearing last timeoutFn of this element? As this may make difference with large app state changes.
```
Comment (mohamedGamalAbuGalala) [inline src/setHighlight.ts:6]: @medhatdawoud 
What do you think? :thinking:

**Review by mohamedGamalAbuGalala — COMMENTED** [review]
Review and planning for the next task.

File: package.json [hunk 3]
```diff
@@ -5,6 +5,12 @@
     "name": "Medhat Dawoud",
     "email": "medhatdawoud@gmail.com"
   },
+  "contributors": [
```
Comment (medhatdawoud) [inline package.json:8]: don't add this here, I'm using all contributors bot to add contributors, you shouldn't add yourself, I'll add you after merging some contributions from you

File: src/setHighlight.ts (hunk 2 above)
Comment (medhatdawoud) [reply src/setHighlight.ts:6]: This is outdated, try getting latest and check code

File: src/createShadowDOM.ts (hunk 1 above)
Comment (medhatdawoud) [reply src/createShadowDOM.ts:18]: No need for that, too much customization, again get latest and check code again, it has been changed

File: src/ChangeHighlight.tsx [hunk 4]
```diff
@@ -25,6 +25,7 @@ const ChangeHighlight: React.FC<Props> = ({ children, hideAfter }) => {
       {shadowDOM.map((child, index) => (
         <ShadowChild key={index} child={child} hideAfter={hideAfter} />
       ))}
+      //TODO: Replace it with childrenWithRef after manipulation.
```
Comment (medhatdawoud) [inline src/ChangeHighlight.tsx:28]: I'm thinking about a similar thing but this now is kind of `shadowDOM` I created, I don't like it much but if you want to try doing that, create a PR and let's see

File: package.json [hunk 5]
```diff
@@ -17,8 +23,8 @@
     "/build"
   ],
   "repository": {
-    "type" : "git",
-    "url" : "https://github.com/medhatdawoud/react-change-highlight"
+    "type": "git",
+    "url": "https://github.com/medhatdawoud/react-change-highlight"
```
Comment (medhatdawoud) [inline package.json:27]: Try to use `prettier` to have the same formatting, just install plugin into your IDE or Editor, and use the ready config one

**Review by medhatdawoud — COMMENTED** [review]
These are some new to have stuff, I prefer to focus more on the open tickets in the main repo, It's more important in the time being, and if you have any suggestion, open an issue with it right away and pick it up in a new PR

File: package.json (hunk 5 above)
Comment (mohamedGamalAbuGalala) [reply package.json:27]: I already use  prettier and didn't notice this change 👌👌

File: src/ChangeHighlight.tsx (hunk 4 above)
Comment (mohamedGamalAbuGalala) [reply src/ChangeHighlight.tsx:28]: I don't see now any other way to do such logic.
About SHADOWDOM, It is good enough for now but still have some cases need to be handled 👌👌

**Pull request closed without merging**